src/core/Str.pm

```diff
@@ -2093,9 +2093,16 @@ multi sub infix:<~>(str $a, str $b) returns str { nqp::concat($a, $b) }
 multi sub infix:<~>(*@args) returns Str:D { @args.join }
 
 multi sub infix:<x>(Str:D $s, Int:D $repetition) returns Str:D {
-    nqp::if(nqp::islt_i($repetition, 0),
+    my Int $max = int.Range.max;
+    nqp::if(
+        nqp::islt_I(nqp::decont($repetition), 0),
         '',
-        nqp::p6box_s(nqp::x(nqp::unbox_s($s), nqp::unbox_i($repetition))))
+        nqp::if(
+            nqp::isgt_I(nqp::decont($repetition), nqp::decont($max)),
+            nqp::p6box_s(nqp::x(nqp::unbox_s($s), nqp::unbox_i($max))),
```

Comment:
This will silently truncate the value, which seems less than great. I'd expect `unbox_i` would throw over a certain size (when it won't unbox into a 64-bit integer), and strings are limited to a 32-bit number of graphemes in MoarVM at present so I'd expect it to fail at that point. If it's not doing so, we should fix it there, I think.